pkg/applicationserver/io/packages/lora-cloud-device-management-v1/api/client.go

```diff
@@ -0,0 +1,109 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+
+	"go.thethings.network/lorawan-stack/pkg/version"
+)
+
+// Option is an option for the API client.
+type Option interface {
+	apply(*Client)
+}
+
+// OptionFunc is an Option implemented as a function.
+type OptionFunc func(*Client)
+
+func (f OptionFunc) apply(c *Client) { f(c) }
+
+// Client is an API client for the LoRa Cloud Device Management v1 service.
+type Client struct {
+	token string
+	cl    *http.Client
+
+	Tokens  *Tokens
+	Uplinks *Uplinks
+}
+
+const (
+	baseURL     = "https://dms.loracloud.com/api/v1"
+	contentType = "application/json"
+)
+
+var (
+	userAgent = "ttn-lw-application-server/" + version.TTN
+)
+
+type queryParam struct {
+	key, value string
+}
+
+func (c *Client) newRequest(method, category, entity, operation string, body io.Reader, queryParams ...queryParam) (*http.Request, error) {
+	u, err := url.Parse(baseURL)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to parse base URL : %v", err))
```

Comment:
You probably want to do this in `init()`, i.e. for performance not parse on every request, and also not lazy-panic. And remove the space between URL and :